finder-app/writer.c

```diff
@@ -0,0 +1,65 @@
+/*
+* File: writer.c 
+* Purpose: Source file for application that writes a string to a file at the system level.
+* Author: Malcolm McKellips
+* Notes:  Assumes that directory structure of file to write has already been created by the caller.
+*
+*/
+
+
+
+#include <stdio.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <dirent.h>
+#include <errno.h>
+#include <syslog.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+int main(int argc, char *argv[]){
+	
+	if (argc < 2){
+		printf("Error, no args provided. Please retry with the following 2 args: [path to file] [string to write]\r\n");
+		syslog((LOG_USER | LOG_ERR),"Error, no args provided. Please retry with the following 2 args: [path to file] [string to write]");
+		return 1;
+	}
+	
+	if (argc == 2){
+		printf("Only first argument (path of file to write) specified. Please also specify second argument (string to write)\r\n");
+		syslog((LOG_USER | LOG_ERR),"Only first argument (path of file to write) specified. Please also specify second argument (string to write)");
+		return 1;	
+	}
+	
+	int fd = -1; //init to -1 same as err so will be caught if not changed
+	char *writefile = argv[1];
+	char *writestr = argv[2];
```

Comment:
Nice plan to use variables here.  What do we know about modification of these variables?  How could the compiler help us enforce this?